lib/fizzy/parser_expr.cpp

```diff
@@ -76,6 +91,23 @@ void update_caller_frame(ControlFrame& frame, const FuncType& func_type)
     frame.stack_height += stack_height_change;
 }
 
+void validate_result_count(const ControlFrame& frame)
```

Comment:
Why isn't this a helper or a constructor on `ControlFrame`?

Reply:
Conceptually: because `ControlFrame` is pure data type, and it does not know how to do validation.
Also: there is `update_caller_frame()`.